integration/proxy_tunnel_strategy_test.go

```diff
@@ -0,0 +1,422 @@
+	privateKey, publicKey, err := testauthority.New().GenerateKeyPair("")
+	require.NoError(t, err)
+
+	auth := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		Priv:        privateKey,
+		Pub:         publicKey,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	auth.AddUser(p.username, []string{p.username})
+
+	conf := service.MakeDefaultConfig()
+	conf.DataDir = t.TempDir()
+	conf.Auth.Enabled = true
+	conf.Auth.NetworkingConfig.SetTunnelStrategy(strategy)
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = false
+
+	require.NoError(t, auth.CreateEx(t, nil, conf))
+
+	t.Cleanup(func() {
+		require.NoError(t, auth.StopAll())
+	})
+	require.NoError(t, auth.Start())
+
+	if p.auth == nil {
+		p.auth = auth
+	} else {
+		t.Error("auth already initialized")
+	}
+}
+
+// makeProxy boostraps a new teleport proxy instance.
+// It's public address points to a load balancer.
+func (p *proxyTunnelStrategy) makeProxy(t *testing.T) {
+	proxy := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	authAddr := utils.MustParseAddr(net.JoinHostPort(p.auth.Hostname, p.auth.GetPortAuth()))
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, *authAddr)
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.SSH.Enabled = false
+
+	conf.Proxy.Enabled = true
+	conf.Proxy.WebAddr.Addr = net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt()))
+	conf.Proxy.PeerAddr.Addr = net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt()))
+	conf.Proxy.PublicAddrs = append(conf.Proxy.PublicAddrs, utils.FromAddr(p.lb.Addr()))
+	conf.Proxy.DisableWebInterface = true
+
+	require.NoError(t, proxy.CreateEx(t, nil, conf))
+	p.lb.AddBackend(conf.Proxy.WebAddr)
+
+	t.Cleanup(func() {
+		require.NoError(t, proxy.StopAll())
+	})
+	require.NoError(t, proxy.Start())
+
+	p.proxies = append(p.proxies, proxy)
+}
+
+// makeNode boostraps a new teleport node instance.
+// It connects to a proxy via a reverse tunnel going through a load balancer.
+func (p *proxyTunnelStrategy) makeNode(t *testing.T) {
+	node := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, utils.FromAddr(p.lb.Addr()))
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = true
+
+	require.NoError(t, node.CreateEx(t, nil, conf))
+
+	t.Cleanup(func() {
+		require.NoError(t, node.StopAll())
+	})
+	require.NoError(t, node.Start())
+
+	if p.node == nil {
+		p.node = node
+	} else {
+		t.Error("node already initialized")
+	}
```

Comment:
Same: check for `p.node != nil` at the beginning of the function and change from `t.Error` to `require.Fail`.